### ISSUE TITLE
Delete File(s) button in Mass Editor

### ISSUE THREAD

**Issue by delize** [Sonarr/Sonarr#5110]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

Lacking flexibility on the deletion of library items

### Describe the solution you'd like

I am currently trying to clean up "old" shows that I have either watched, grabbed (but have no intention of watching), or simply don't want. 

However I would like to filter or make assumptions on (meta) data based on Sonarr information. Things like ratings, release date, etc. 

Normally, I could delete an entire folder on the disk itself, but, then I wouldn't be able to make assumptions or determine specifications with the files I would want to delete (EG: it would be more work this way). 

### Describe alternatives you've considered

Filebot
Removing files by hand

### Anything else?

In the mass editor, click the delete button branch out a specification of "Delete series and files" and another option of "Delete files only".

**Comment by bakerboy448** [comment]
Why would you delete all episodes for a series, but not delete the series from Sonarr?  That seems strange....

Otherwise, you already have the option to delete files when deleting series - even in bulk from mass editor.
![image](https://user-images.githubusercontent.com/55419169/183480836-f6e8cb6a-2931-4360-a9a2-7a53fc5c2689.png)

**Comment by delize** [comment]
When wanting to keep track of what you have grabbed or series that you may want to grab in the future. 

Especially if you use Sonarr as a tracking service for what has been added. The DB is a helpful/useful resource even if the files are no longer there.

**Comment by bakerboy448** [comment]
Well if you'd still need to go an unmonitor all the historical seasons and episodes if you delete the files and not the series, so this is already a multi step process.

Current solution for that would be:
- delete series and files via mass editor
- add series back to sonarr, monitor future only (or monitor none)

**Comment by delize** [comment]
Both of those ways seem to be a multi step process.

For what it is worth, I am not dismissing that it may end up that way, but it seems like having to re-add every series is less convenient and more time consuming (and also imposes more API checks / traffic) as opposed to:

1.  checking a bunch of boxes
2. clicking "Unmonitor". 
3. Then using the  filter for "Unmonitored" and
4.  then selecting the checkbox in the top left 
<img width="308" alt="image" src="https://user-images.githubusercontent.com/4028612/183482882-28eb75d8-3160-4923-9985-ed5927056aef.png">
5. Clicking "Delete files"

That seems like a lot less steps then delete series and files via mass editor, re-add any potential series (and if it is more than one series, you already are doing less steps using the above method).

**Comment by markus101** [comment]
Deleting the series would also remove the history of it.

If you're just trying to track things you've removed, adding it to the exclusion list (which could also be indicated on the add series screen, not currently there, but something Radarr shows).

> However I would like to filter or make assumptions on (meta) data based on Sonarr information. Things like ratings, release date, etc.

For this you're already going to need to go to series details page, so couldn't you just remove all the episode files through Manage Episodes?

**Comment by delize** [comment]
> For this you're already going to need to go to series details page, so couldn't you just remove all the episode files through Manage Episodes?

No, this could be a filter? Why would I need to go to the series detail page?
<img width="825" alt="image" src="https://user-images.githubusercontent.com/4028612/183483710-9e951327-ac92-426a-b411-fa41112ed343.png">


You would then utilize that Filter in the Mass Editor.

**Comment by bakerboy448** [comment]
> I am currently trying to clean up "old" shows that I have either watched, grabbed (but have no intention of watching), or simply don't want.

and

> When wanting to keep track of what you have grabbed or series that you may want to grab in the future.

These are contradictory. 

If you don't care about a series - don't add it to sonarr.
If you do care about a series - add it to sonarr and monitor it as you desire.

> Especially if you use Sonarr as a tracking service for what has been added. 

Don't think sonarr has ever been advertised as any sort of "tracking service"  and a series simply being added and existing in sonarr is meaningless and tells one nothing by itself.,

**Comment by delize** [comment]
These two are not mutually exclusive and can exist in a contradictory sense. "Caring about a series" and "Running out of disk space" are two different issues. I am not deleting files because "I don't care about the series", but I am deleting them because I need to free up disk space on a rotating basis.

1. I care about watching series X  at one point in time. 
2. I set up series X to watch.
3. Series X ends.
4. I watch all of Series X and want to remove it to free up disk space but still care about the series for the future.
5. 8 months later, one of two things happen
-  I get an urge to watch Series X but I can't remember the exact name of the series so I can't re-add it to Sonarr
-  I browse Sonarr and see the series name and re-grab all the episodes when I want to watch it.


It provides flexibility with disk space, while still having a record of what has been grabbed, history for what has been grabbed, and you save all the meta information related to what you had in terms of previous pulls.

This allows you to maximize disk space for local disk storage, while still retaining care of the series in Sonarr.



>> keep the history of what has been grabbed from the series, but can clear up files on disk

> what information and benefit does this provide and what actions can be taken on it?

See above. It is a manual process, but, when you have 1000+ series, manual actions on data that is retained in Sonarr seem to be the easiest method of achieving clean up.

**Comment by delize** [comment]
> Plex has an option to delete after watching - per series - and Sonarr has the corresponding option to unmonitor deleted episodes.

This is not only clunky given Plex's interface, it also still has its own issues. You need to stay up to date on new series that are added at any given point (which if using automated import lists for Trakt based on trending content or anticipated), is difficult - even with filtering out specific top results.


> So history of a series doesn't matter from what you've described

How are you coming to that conclusion? I have explained several times how the history of a series does matter. I am not sure what you are searching for in terms of "what matters" but the meta information of where it is grabbed and what file/release was grabbed is still helpful because I could grab that same release if the quality was good, or if there is a better quality, it could be compared. 


> Perhaps you should explore other services - like Trakt - that are specifically designed to track what you watch. That's not Sonarr's purpose.

You mentioned Trakt, but I have already mentioned Trakt[ a few posts back](https://github.com/Sonarr/Sonarr/issues/5110#issuecomment-1208439435) - yes, I do use it, but I find it's services to be less than desirable with some limitations under the free plan. Especially when some of those same functions already exist directly in Sonarr (either purposefully or not) and can be enhanced via minimal tweaks. (Like this suggestion)

**Comment by bakerboy448** [comment]
> the meta information of where it is grabbed and what file/release was grabbed is still helpful because I could grab that same release if the quality was good, or if there is a better quality, it could be compared.

Historical data has really no relevance to sonarr if the episode is missing.  You cannot necessarily grab the same release as it may no longer exist.  Each search is independent and does not consider history...only on-disk files and any blocklisted releases.

So you're saying when you go to monitor and search the "deleted" show's episodes that you go look at the history for every episode and then search for that specific release or tell sonarr to search and then go look at the history for each episode and compare it to what was grabbed?  It sounds like you'd be better off configuring your release profiles correctly to target and grab the releases you want.

**Comment by markus101** [comment]
> So I think we're still back the the main question of what specific benefit does adding a button and all the required logic and api calls to delete all episodes in multiple series via mass editor add that can't already be accomplished through existing methods add?

Pretty much everything you can do via mass editor is possible via other means, deleting the files is pretty niche, but I could also see someone wanting to wipe out files that are mapped completely wrong and they just want to start fresh, but maintain the blocklist or other history.

I definitely see this as a niche feature that a lot of people might not use and may take a while for someone to get around to it, but overall something that wouldn't be incredibly complicated to add and I don't think needs such a level of scrutiny.

**Comment by undaunt** [comment]
I agree, my use case is moving large amounts of data between ZFS data sets or different NAS solutions where it is far faster to redownload the content when I need to juggle available HDDs instead of having to rsync it or go into each series one at a time to mass delete the files with the intent to redownload the series in the future.

Keeping the series in Sonarr but only deleting the files, en masse, would be a huge mass editor value add.

**Comment by seraphx2** [comment]
The close-mindedness of some people is staggering. Developers should know better (I am one, so have experience to say this). It's one thing to need to verify/question a workflow to determine if it's worth coding but to act like you understand all use cases and all others are beneath you or that "your way is the only way" to use a product is super elitist and gives off a very condescending attitude.

Sonarr is literally a file management tool except all it does well is download files. I think the -arr stack should most definitely be involved in helping manage disk space as they are the ones helping fill it up. There is an issue in here from 2013 where people have requested automatic "rolling deletions" based on a time frame, per series (I kinda like this idea as well). Making it easier to manage disk space really should be a consideration of this app and how someone can't see how this would be immensely useful is beyond me.

And in this request, it's just some freakin filters that would help delete episodes and would be pretty easy to implement and on the onus of the user to do it. I think automation is a bad idea, but deliberate interaction by the user makes much more sense here.

I could have really used this feature several times now and it would have made my life much easier.

Though, I would say this should be a feature in both the Mass Editor and on the page of each series since deleting individual episodes on the series page is not an easy task. Or you can't delete even a whole season for that mater (what?).

Also, saying "go rely on Plex" is a dumb solution because not everyone uses Plex. Plexe is a video player, not a file management system. You're talking about what apps should handle what and Plex is way more divorced from the nature of this request than Sonarr is.

**Comment by markus101** [comment]
This is quite an aggressive way to join the conversation.

> And in this request, it's just some freakin filters that would help delete episodes and would be pretty easy to implement and on the onus of the user to do it.

The filters already exist, it's a matter of adding the functionality to delete them. Since Sonarr is open source any developer could contribute this functionality and given it's limited scope a great first issue for someone.

> Though, I would say this should be a feature in both the Mass Editor and on the page of each series since deleting individual episodes on the series page is not an easy task.

The latter already exists, under Manage Episodes.

> Or you can't delete even a whole season for that mater (what?).

You can already do this at the series or season level from the series page.

![image](https://github.com/Sonarr/Sonarr/assets/413544/79ebce2c-d260-4321-a46c-3427c4570e44)

> Also, saying "go rely on Plex" is a dumb solution because not everyone uses Plex. Plexe is a video player, not a file management system. 

No, but it has the functionality to delete files after being watched, which is a common reason people want to remove files automatically, Emby and others likely have similar functionality.

**Comment by seraphx2** [comment]
Sorry, I suppose I did come in hard, but attitudes like bakerboy's always gets on my nerves. He wasn't critiquing the idea for feasibility, he was shooting it down, outright. Now, if he had tried to explain it like you did, I would have said nothing. The way you approached it is the correct way. He also seems to come at this from an individual household perspective, but as a service for friends, you may need to use more than just delete the file from Plex after watched. If you go look at [Cleanarr](https://github.com/Cleanarr/Cleanarr), it has some interesting ideas for specifying rules on how to automatically handle cleanup.

> The filters already exist, it's a matter of adding the functionality to delete them. Since Sonarr is open source any developer could contribute this functionality and given it's limited scope a great first issue for someone.

This is at the Series level on the Mass Editor, which does not account for episodes, so that would be an entirely different lift and would not be a simple implementation. Something like this is a fundamental change to the philosophy of what has been implemented so far and I am sure would require buy-in from the main developers before just merging it in so it would make sense to talk it out and get approval before taking the time to code it. And I say that seeing as how this app has been around a long time but has no ability to mass edit episodes across series at the same time.

> The latter already exists, under Manage Episodes.

Heh, so my bad on that one. I forgot there is that option on that screen per season. I guess that's not too bad to handle it that way. But definitely only good at one season at a time, one series at a time. It would be much easier if you're trying to mass delete based on criteria to be able to do it across series. Write a filter, see what it would affect, be able to uncheck things you don't want it to delete, click button. The whole idea here is to more readily be able to free up space. When you're upwards of 100 shows, that could be a pain.

I said per series, but maybe that's not needed. I think the Mass Editor would be a good place to implement this, but that location has, to this day, only been about mass editing series' metadata, not episodes. It can delete, but once again, at a series level: all or nothing.

**Comment by enoch85** [comment]
Ok, so 2 years passed... Any development in this matter?
